webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyService.java

```diff
@@ -126,6 +131,9 @@ void syncMojitoWithThirdPartyTMS(Long repositoryId,
         if (actions.contains(ThirdPartySyncAction.PUSH_SCREENSHOT)) {
             uploadScreenshotsAndCreateMappings(repository, thirdPartyProjectId);
         }
+
+        MDC.remove(REPOSITORY_ID);
+        MDC.remove(THIRD_PARTY_PROJECT_ID);
```

Comment:
i'm not a fan of having to remove manually here. an Aspect would be great for that use case

Reply:
or the `putCloseable`?